### DISCUSSION TITLE
keyball39 y,h,n キーが全て反応しない

### DISCUSSION THREAD

**Discussion by fukushima-shoma** [Yowkees/keyball#416] [Q&A]
お世話になります。

現在 Build Guide for Keyball39 に従い組み立てを進めております。

機種：Keyball39 ver2
状況 :
ステップ [3-7．ProMicroの書き込みと動作確認](https://github.com/Yowkees/keyball/blob/main/keyball39/doc/rev1/buildguide_jp.md#3-7promicro%E3%81%AE%E6%9B%B8%E3%81%8D%E8%BE%BC%E3%81%BF%E3%81%A8%E5%8B%95%E4%BD%9C%E7%A2%BA%E8%AA%8D) にて、
症状1 :
キースイッチにピンセットを当て入力確認を行っているのですが、特定のキーのみ確認ができません。 (添付写真箇所)

縦一列すべてのキーが反応しないため、ProMicro裏の半田付けが原因かと考えているのですが、該当キーと連動するProMicroピンについてご教授いただきたいです。

よろしくお願いいたします。
![IMG_8741 大](https://github.com/Yowkees/keyball/assets/54273006/0eb9d696-5f68-4970-8b00-2434af693482)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

縦一列ですので、COL4という信号線があやしいですが、そうなると下記画像の黄色丸のキーも効かないはずですがどうでしょうか？

COL4はProMicro裏の緑点で示した端子２つです。
これが他とショートしていないか？またProMicro本体まできちんと導通しているかご確認ください。
また緑点で示した部分がすべてCOL4です。これらがすべて導通し、基板上で断線等していないかご確認いただければ幸いです。

以上、お手数ですがご確認お願いいたします。

![KB39_tatefix](https://github.com/Yowkees/keyball/assets/84209772/98d2ead4-c6c1-4748-a7d2-50ba9b2a7f54)

**Reply by fukushima-shoma** [reply]
ご確認いただきありがとうござます。

>縦一列ですので、COL4という信号線があやしいですが、そうなると下記画像の黄色丸のキーも効かないはずですがどうでしょうか？

ご指摘の通り、黄色丸のキーも効かない状況です。

>これが他とショートしていないか

目視で確認した限りでは、他とショートしないように見えます。

>またProMicro本体まできちんと導通しているかご確認ください。

緑点で示していただい箇所が導通していることは確認できました。

>また緑点で示した部分がすべてCOL4です。これらがすべて導通し、基板上で断線等していないかご確認いただければ幸いです。

すべて導通していることを確認しました。

上記、ご案内いただいた項目はチェックしたのですが、症状の改善はない状況となっております。
この場合、他に原因として確認すべき項目はございますでしょうか？

ご確認のほどよろしくお願い申し上げます。

**Comment by Yowkees** [comment]
ご確認ありがとうございます。

COL4線の導通に問題がないようでしたら、ProMicroに原因があるかも知れません。左右のProMicroを入れ替えるとどうでしょうか？
その際に12Pピンソケット内に半田が流れ込んでいないかもご確認いただければ幸いです。

以上、お手数ですがご確認お願いいたします。
